src/storage/storage_manager.cpp

```diff
@@ -117,45 +137,49 @@ void SingleFileStorageManager::LoadDatabase() {
 	options.read_only = read_only;
 	options.use_direct_io = config.options.use_direct_io;
 	options.debug_initialize = config.options.debug_initialize;
+
 	// first check if the database exists
 	if (!fs.FileExists(path)) {
 		if (read_only) {
 			throw CatalogException("Cannot open database \"%s\" in read-only mode: database does not exist", path);
 		}
+
 		// check if the WAL exists
+		auto wal_path = GetWALPath();
 		if (fs.FileExists(wal_path)) {
 			// WAL file exists but database file does not
 			// remove the WAL
 			fs.RemoveFile(wal_path);
 		}
+
 		// initialize the block manager while creating a new db file
 		auto sf_block_manager = make_uniq<SingleFileBlockManager>(db, path, options);
 		sf_block_manager->CreateNewDatabase();
 		block_manager = std::move(sf_block_manager);
 		table_io_manager = make_uniq<SingleFileTableIOManager>(*block_manager);
+
 	} else {
 		// initialize the block manager while loading the current db file
 		auto sf_block_manager = make_uniq<SingleFileBlockManager>(db, path, options);
 		sf_block_manager->LoadExistingDatabase();
 		block_manager = std::move(sf_block_manager);
 		table_io_manager = make_uniq<SingleFileTableIOManager>(*block_manager);
 
-		//! Load from storage
-		auto checkpointer = SingleFileCheckpointReader(*this);
-		checkpointer.LoadFromStorage();
+		// load the db from storage
+		auto checkpoint_reader = SingleFileCheckpointReader(*this);
+		checkpoint_reader.LoadFromStorage();
+
 		// check if the WAL file exists
+		auto wal_path = GetWALPath();
 		if (fs.FileExists(wal_path)) {
 			// replay the WAL
-			truncate_wal = WriteAheadLog::Replay(db, wal_path);
-		}
-	}
-	// initialize the WAL file
-	if (!read_only) {
-		wal = make_uniq<WriteAheadLog>(db, wal_path);
-		if (truncate_wal) {
-			wal->Truncate(0);
+			if (WriteAheadLog::Replay(db, wal_path)) {
+				fs.RemoveFile(wal_path);
```

Comment:
@Vegetable26 @jkub this changes the behavior from truncating to removing the WAL file entirely if it is detected that a crash has occurred right after checkpointing and the WAL itself does not need to be used anymore, does that cause any problems on your end?